### PR TITLE
Add remaining G Suite syncing.

### PR DESCRIPTION
**Pull request by AlbertoPdRF**
I haven't replicated the tests from #3604, let me know if I should!

### CODE REVIEW

File: WcaOnRails/app/jobs/sync_mailing_lists_job.rb [hunk 1]
```diff
@@ -4,8 +4,20 @@ class SyncMailingListsJob < ApplicationJob
   queue_as :default
 
   def perform
+    GsuiteMailingLists.sync_group("candidatedelegates@worldcubeassociation.org", User.where(delegate_status: "candidate_delegate"))
```
Comment (jfly) [inline WcaOnRails/app/jobs/sync_mailing_lists_job.rb]: Maybe add a dot or a dash in this mailing list name?

File: WcaOnRails/app/jobs/sync_mailing_lists_job.rb [hunk 2]
```diff
@@ -4,8 +4,20 @@ class SyncMailingListsJob < ApplicationJob
...
     GsuiteMailingLists.sync_group("delegates@worldcubeassociation.org", User.delegates)
+    GsuiteMailingLists.sync_group("seniors@worldcubeassociation.org", User.where(delegate_status: "senior_delegate"))
```
Comment (jfly) [inline WcaOnRails/app/jobs/sync_mailing_lists_job.rb]: Could you add a scope to the Users model for this and candidate delegates (if we don't already have them). I'd prefer that these strings not leak out of the user's model.

**Comment by jfly** [comment]
Thanks for doing this, Alberto! Did you check if that test still passes? I expect it will fail now because you changed the method and it's now calling sync with unexpected arguments. I haven't checked, but I'd guess that's why travis is failing.

File: WcaOnRails/app/jobs/sync_mailing_lists_job.rb (hunk 2 above)
Comment (AlbertoPdRF) [reply WcaOnRails/app/jobs/sync_mailing_lists_job.rb]: Sure thing!

**Comment by AlbertoPdRF** [comment]
Indeed, the test fails and I'm fixing it now.

**Comment by AlbertoPdRF** [comment]
Should be ok now, let's see!

**Pull request merged**